### DISCUSSION TITLE
JSVM middleware example to read and rewrite response body?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5996] [Q&A]
Does anyone know how to do this? I'd like to create a `routerUse` middleware that can read and alter the response body before sending it along.

**Comment by ganigeorgiev** [comment]
Please be more specific about your use case and what you are trying to do.

There is no hook that will intercept the response body writes. What you can do is assign a custom/wrapped `ResponseWriter` implementation (aka. changing the `e.Response` field) but this can be done only with Go.

**Reply by benallfree** [reply]
My use case was to rewrite the admin `<meta>` tag mentioned in #5995. There might be other cases where the rewrite would be convenient for making the plugin work more seamlessly.

**Reply by ganigeorgiev** [reply]
You can technically intercept the request based on the url path and return a response write with a custom html BEFORE the default static file serve executes (aka. before calling next), but there are no plans to allow intercepting the response write itself from the JSVM as there are a lot of edge cases and this is better done on Go level by properly implementing a custom `ResponseWriter`.
